develop/getting-started/using-the-ide.md

```diff
@@ -0,0 +1,246 @@
+If you closed them by accident or just moved your cursor to a new place, you can use `CTRL + P` to open them up again.
+
+Methods and classes can have multiple implementations with different parameters, also known as Overloading. This way you can decide on which
+implementation you want to use, while writing the method call.
+
+![Displaying method parameters](/assets/develop/getting-started/using-the-ide/util_02.png)
+
+### Refactoring
+
+Refactoring is the process of restructuring code without changing its runtime functionality. Renaming and Deleting parts of the code safely is a part of that,
+but things like extracting parts of the code into separate methods and introducing new variables for repeated code statements are also called "refactoring".
+
+Many IDEs have an extensive tool kit to aid in this process. In IntelliJ simply right click files or parts of the code to get access to the available refactoring tools.
+
+![Refactoring](/assets/develop/getting-started/using-the-ide/refactoring_01.png)
+
+It is especially useful to get used to the `Rename` refactoring tool's key bind, `SHIFT + F6`, since you will rename many things in the future. Using this feature,
+every code occurrence of the renamed code will be renamed and will stay functionally the same.
+
+### Search and Replace file content
+
+Sometimes simpler tools are needed to edit code occurrences.
+
+| Key bind           | Function                                                    |
+| ------------------ | ----------------------------------------------------------- |
+| `CTRL + F`         | Find in current file                                        |
+| `CTRL + R`         | Replace in current file                                     |
+| `CTRL + SHIFT + F` | Find in a bigger scope (can set specific file type mask)    |
+| `CTRL + SHIFT + R` | Replace in a bigger scope (can set specific file type mask) |
+
+If enabled, all of those tools allow for a more specific pattern matching using "[Regex](https://en.wikipedia.org/wiki/Regular_expression)".
+
+![Regex replace](/assets/develop/getting-started/using-the-ide/search_and_replace_01.png)
+
+## Comments
+
+Good code should be easily readable and [self-documenting](https://bytedev.medium.com/code-comment-anti-patterns-and-why-the-comment-you-just-wrote-is-probably-not-needed-919a92cf6758). Picking expressive names for variables, classes and methods can help a lot, but sometimes
+comments are necessary to leave notes or **temporarily** disable code for testing.
+
+### Prepare Shortcuts
+
+To comment out code faster, open IntelliJ's Settings, look for the `Comment with Line Comment`
+and the `Comment with Block Comment` entries, and set their Key binds to your preferences.
+
+![Keymap settings](/assets/develop/getting-started/using-the-ide/comments_01.png)
+
+Now you can highlight the necessary code and use the shortcuts, to comment the section out.
+
+```java
```

Comment:
What is the purpose of these code blocks, im not sure that they're really needed?

Reply:
They show how (multiline) comments look like and how to use them.
Lmk if i should remove them.

Reply:
Yeah I dont think you need a codeblock to emphasise this, most people know what a comment is :stuck_out_tongue_winking_eye: